lib/tasks/data_hygiene.rake

```diff
@@ -32,4 +32,18 @@ namespace :data_hygiene do
       speech.save!(validate: false)
     end
   end
+
+  desc "Restore a deleted document by creating a new edition"
+  task :restore_deleted_document, %i[document_id user_email] => :environment do |_, args|
+    deleted_document_restorer = DataHygiene::DeletedDocumentRestorer.new(args[:document_id], args[:user_email])
+
+    begin
+      deleted_document_restorer.run!
+      puts "Created a new draft for document with ID #{args[:document_id]}"
+    rescue DataHygiene::DeletedDocumentRestorer::DocumentStateError
+      puts "Error: This document's most recent edition is not deleted."
+    rescue DataHygiene::DeletedDocumentRestorer::UserNotFound
+      puts "Error: Unable to find user with email #{args[:user_email]}"
```

Comment:
These `puts` statements duplicate what you've captured in the `deleted_document_restorer.rb`. I'd opt for an approach like:

```suggestion
  rescue DataHygiene::DeletedDocumentRestorer::DocumentStateError, DataHygiene::DeletedDocumentRestorer::UserNotFound => e
    puts "Error: #{e.message}"
```

Maybe even have the exception classes inherit from a parent so that the exception-handling code is easier to read:

```suggestion
  rescue DataHygiene::DeletedDocumentRestorer::Exception => e
    puts "Error: #{e.message}"
```